src/charm.py

```diff
@@ -127,6 +129,7 @@ def __init__(self, *args):
                 retention=to_tuple(ActiveStatus()),
             )
         )
+        self._stored.set_default(is_upgrade=False)
```

Comment:
Wdyt about adding a comment such as:
```suggestion
        self._stored.set_default(is_upgrade=False)  # Relies on controller-backed storage
```